degrees.json

```diff
@@ -0,0 +1,114 @@
+[
+  {
+    "id": "informatica",
+    "name": "Informatica",
+    "icon": "🧑‍💻",
+    "years": [
+      {
+        "year": 1,
+        "teachings": [
+          {
+            "$ref": "teachings.json#/1/teachings?filter=id&value=informatica"
+          }
+        ]
+      },
+      {
+        "year": 2,
+        "teachings": [
+          {
+            "$ref": "teachings.json#/2/teachings?filter=id&value=informatica"
+          }
+        ]
+      },
+      {
+        "year": 3,
+        "teachings": [
+          {
+            "$ref": "teachings.json#/3/teachings?filter=id&value=informatica"
+          }
+        ]
+      }
+    ]
+  },
+  {
+    "id": "ingegneria",
+    "name": "Ingegneria Informatica",
+    "icon": "👨‍🔧",
+    "years": [
+      {
+        "year": 1,
+        "teachings": [
+          {
+            "$ref": "teachings.json#/1/teachings?filter=id&value=ingegneria"
+          }
+        ]
+      },
+      {
+        "year": 2,
+        "teachings": [
+          {
+            "$ref": "teachings.json#/2/teachings?filter=id&value=ingegneria"
+          }
+        ]
+      },
+      {
+        "year": 3,
+        "teachings": [
+          {
+            "$ref": "teachings.json#/3/teachings?filter=id&value=ingegneria"
+          }
+        ]
+      }
+    ]
+  },
+  {
+    "id": "informatica-magistrale",
+    "name": "Informatica Magistrale",
+    "icon": "🧑‍🏫",
+    "years": [
+      {
+        "year": 1,
+        "teachings": [
+          {
+            "$ref": "teachings.json#/1/teachings?filter=id&value=informatica-magistrale"
+          }
+        ]
+      }
+    ]
+  },
+  {
+    "id": "ingegneria-e-scienze-informatiche-magistrale",
+    "name": "Ingegneria e Scienze Informatiche Magistrale",
+    "icon": "🥸",
+    "years": [
+      {
+        "year": 1,
+        "teachings": [
+          {
+            "$ref": "teachings.json#/1/teachings?filter=id&value=ingegneria-e-scienze-informatiche-magistrale"
+          }
+        ]
+      }
+    ]
+  },
+  {
+    "id": "artificial-intelligence",
+    "name": "Artificial Intelligence Master's Degree",
+    "icon": "🧠",
+    "years": [
+      {
+        "year": 1,
+        "teachings": [
+          {
+            "$ref": "teachings.json#/1/teachings?filter=id&value=artificial-intelligence"
+          }
+        ]
+      }
+    ]
+  },
+  {
+    "id": "lab",
```

Comment:
Stiamo usando l'id anche come nome della repo in questo caso? Se sì va bene. Volevo solo capire.

Reply:
Sì, avevo cercato di mantenere le cose il più simile a courses. l'ID rappresentava anche lì il _degree_

Reply:
Quando si progettano basi di dati, penso che la chiave primaria e la chiave esterna debbano avere sempre nomi diversi: la chiave primaria non deve contenere il nome della sua struttura perché vi è già dentro, quindi è chiaro dal contesto.  La chiave esterna invece dovrebbe farvi riferimento per chiarire la relazione dal nome.